provider/oracle/environ.go

```diff
@@ -0,0 +1,389 @@
+	return nil
+}
+
+// AdoptResources is called when the model is moved from one
+// controller to another using model migration. Some providers tag
+// instances, disks, and cloud storage with the controller UUID to
+// aid in clean destruction. This method will be called on the
+// environ for the target controller so it can update the
+// controller tags for all of those things. For providers that do
+// not track the controller UUID, a simple method returning nil
+// will suffice. The version number of the source controller is
+// provided for backwards compatibility - if the technique used to
+// tag items changes, the version number can be used to decide how
+// to remove the old tags correctly.
+func (e oracleEnviron) AdoptResources(controllerUUID string, fromVersion version.Number) error {
+
+	return nil
+}
+
+// StartInstance asks for a new instance to be created, associated with
+// the provided config in machineConfig. The given config describes the juju
+// state for the new instance to connect to. The config MachineNonce, which must be
+// unique within an environment, is used by juju to protect against the
+// consequences of multiple instances being started with the same machine
+// id.
+func (o oracleEnviron) StartInstance(args environs.StartInstanceParams) (*environs.StartInstanceResult, error) {
+	if args.ControllerUUID == "" {
+		return nil, errors.NotFoundf("Controller UUID")
+	}
+
+	fmt.Printf("%+v\n", args)
+	os.Exit(1)
+	types, err := getInstanceTypes(o.p.client)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	imagemetadata, err := checkImageList(
+		o.p.client, args.Constraints,
+	)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	//find the best suitable instance returned from the api
+	spec, imagelist, err := findInstanceSpec(
+		o.p.client,
+		imagemetadata,
+		types,
+		&envinstance.InstanceConstraint{
+			Region:      o.spec.Region,
+			Arches:      []string{"amd64"},
+			Constraints: args.Constraints,
+		},
+	)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// we are using just the juju ssh keys, no other keys
+	keys := strings.Split(o.cfg.AuthorizedKeys(), "\n")
+	// we should try to determine if we need to upload the keys or if
+	// there is already the set of keys there if we found the ssh key
+	// then we should check if the key has the flag enabled on true
+	// if not, change it to true
+	// if the key is not present in the oracle infrstracture then we should
+	// make a request and upload it to make use for further bootstraping the
+	// controller or adding other machines
+	nameKey, err := uploadSSHControllerKeys(o.p.client, keys[0])
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	instance, err := launchBootstrapConstroller(o.p.client, oci.InstanceParams{
+		Relationships: nil,
+		Instances: []oci.Instances{
+			{
+				Shape:     spec.InstanceType.Name,
+				Imagelist: imagelist,
+				Name:      o.cfg.Name(),
```

Comment:
I don't think you want to use `o.cfg.Name()`. That's the *model* name. See use of "instance.NewNamespace" in the MAAS provider. Using that you'll get a name like "juju-f33df4c3-0".